src/fsharp/tast.fs

```diff
@@ -574,12 +577,30 @@ type Entity =
       /// Used during codegen to hold the ILX representation indicating how to access the type 
       // MUTABILITY: only for unpickle linkage and caching
       mutable entity_il_repr_cache : CompiledTypeRepr cache
+
+      mutable entity_opt_data : EntityOptionalData option
     }
+
+    static member EmptyEntityOptData = { entity_compiled_name = None; entity_other_range = None; entity_kind = TyparKind.Type; entity_xmldoc = XmlDoc.Empty; entity_xmldocsig = ""; entity_tycon_abbrev = None; entity_tycon_repr_accessibility = TAccess []; entity_accessiblity = TAccess []; entity_exn_info = TExnNone }
```

Comment:
As mentioned on previous PR, using a static property can be misleading when generating a mutable data structure.  Perhaps a static method ``MyEmptyEntityOptionalData()``